### PR TITLE
fix: email permission validations

### PR DESCRIPTION
**Pull request by s-barrah**
fix: email permission validations

### CODE REVIEW

File: tests/unit/Model/SQS/MarketingPreferences.model.js [hunk 1]
```diff
@@ -185,6 +184,8 @@ describe('Model/MarketingPreferencesModel', () => {
           done();
         })
         .catch((error) => {
+          console.log('Error: ', error);
```
Comment (lebaz20) [inline tests/unit/Model/SQS/MarketingPreferences.model.js]: Do we still need that debugging log ?

File: tests/unit/Model/SQS/MarketingPreferences.model.js [hunk 2]
```diff
@@ -168,8 +168,7 @@ describe('Model/MarketingPreferencesModel', () => {
       transSource: 'giftaid-sportrelief',
       transSourceUrl: 'https://giftaid.sportrelief.com/',
       transType: 'prefs',
-      email: '',
```
Comment (lebaz20) [inline tests/unit/Model/SQS/MarketingPreferences.model.js]: from title : 'Ensure validation fails when email permission is set and no email is provided'
I think `email` should be present, and `permissionEmail` should be set to 1, right ?

File: src/Model/SQS/MarketingPreference.model.js [hunk 3]
```diff
@@ -466,14 +466,13 @@ export default class MarketingPreference extends Model {
         && this.getPermissionEmail() !== ''
         && this.getPermissionEmail() !== '0')
       || this.getEmail() !== '') {
-        if (this.getEmail() !== '') {
+        if (this.getEmail() !== '' && this.getEmail() !== null) {
```
Comment (lebaz20) [inline src/Model/SQS/MarketingPreference.model.js]: can we use this ?
```suggestion
        if (this.getEmail()) {
```

File: tests/unit/Model/SQS/MarketingPreferences.model.js (hunk 2 above)
Comment (s-barrah) [reply tests/unit/Model/SQS/MarketingPreferences.model.js]: No, we want validation to fail when no email is provided but the permission is set to `1` or `0`

**Pull request merged**